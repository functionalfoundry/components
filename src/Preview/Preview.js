/* @flow */
import React from 'react'
import {
  Fonts,
  Spacing,
} from '@workflo/styles'
import View from '../View'

type PropsT = {
  label: string,
  children?: React.Children,
}

const defaultProps = {
  label: '',
}

const Preview = ({
  label,
  children,
}: PropsT) => (
  <View
    theme={{
      view: previewStyle,
    }}
  >
    {label &&
      <View
        theme={{
          view: labelStyle,
        }}
      >
        {label}
      </View>}
    {children &&
      <View>
        {children}
      </View>}
  </View>
)

const previewStyle = {
  marginBottom: Spacing.base,
  flex: '0 1',
}

const labelStyle = {
  ...Fonts.base,
}

Preview.defaultProps = defaultProps
export default Preview
