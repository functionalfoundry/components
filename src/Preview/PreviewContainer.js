/* @flow */
import React from 'react'
import {
  Colors,
} from '@workflo/styles'
import View from '../View'

type ShadeT = 'light' | 'dark'

type PropsT = {
  color: string,
  children?: React.Children,
  flush: boolean,
  shade: ShadeT,
}

const defaultProps = {
  flush: false,
  shade: 'dark',
}
// HACK: For storybook
document.getElementsByTagName('body')[0].setAttribute('style', 'margin:0;')

const Preview = ({
  children,
  color,
  flush,
  shade,
}: PropsT) => (
  <View
    theme={{
      view: containerStyle(color, flush, shade),
    }}
  >
    {children}
  </View>
)

const containerStyle = (color: string, flush: boolean, shade: ShadeT) => ({
  height: '1000px',
  ...getColorStyle(color, shade),
  ...getFlushStyle(flush),
  justifyContent: 'flex-start',
  alignItems: 'flex-start',
})

const getColorStyle = (color: string, shade: ShadeT) => {
  if (color) return { backgroundColor: color }
  switch (shade) {
    case 'light':
      return {
        backgroundColor: Colors.grey200,
      }
    case 'dark':
      return {
        backgroundColor: Colors.grey900,
        // THINK: Is setting color on the Preview too big of a side effect?
        color: Colors.grey100,
      }
    default:
      return {}
  }
}

const getFlushStyle = (flush: boolean) => {
  if (flush) {
    return {}
  }
  return {
    padding: 20,
  }
}

Preview.defaultProps = defaultProps
export default Preview
