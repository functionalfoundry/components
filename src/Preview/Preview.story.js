import React from 'react'
import { storiesOf } from '@kadira/storybook'
import Preview from '.'
import PreviewContainer from './PreviewContainer'
import Button from '../Button'

storiesOf('Preview', module)
  // .add('Regular', () => (
  //   <div
  //     style={styles.container}
  //   >
  //     <Preview
  //       label='Regular'
  //     >
  //       <Button>{'Hello'}</Button>
  //     </Preview>
  //     <Preview
  //       label='Ghost'
  //     >
  //       <Button ghost>{'Click'}</Button>
  //     </Preview>
  //   </div>
  // ))
  // .add('Container', () => (
  //   <PreviewContainer
  //     color='#888'
  //   >
  //     {'Text'}
  //   </PreviewContainer>
  // ))

// const styles = {
//   container: {
//     width: '100%',
//   },
// }
