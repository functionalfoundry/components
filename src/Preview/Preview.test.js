import React from 'react'
import renderer from 'react-test-renderer'
import Preview from '.'

test('Renders without props', () => {
  const preview = renderer.create(
    <Preview />
  ).toJSON()
  expect(preview).toMatchSnapshot()
})
