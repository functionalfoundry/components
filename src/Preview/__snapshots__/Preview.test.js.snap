exports[`test Renders without props 1`] = `
<div
  className="view_1ps7m7i preview_120drhm"
  data-test-id={undefined} />
`;
